File: src/flink-examples/flink-examples-security/FlinkKafkaScalaExample/src/main/java/com/huawei/bigdata/flink/examples/WriteIntoKafkaUseKafkaSink.scala
package com.huawei.bigdata.flink.examples

import org.apache.flink.api.common.serialization.{SerializationSchema, SimpleStringSchema}
import org.apache.flink.api.java.utils.ParameterTool
import org.apache.flink.connector.kafka.sink.{KafkaRecordSerializationSchema, KafkaSink}
import org.apache.flink.streaming.api.functions.source.SourceFunction
import org.apache.flink.streaming.api.functions.source.SourceFunction.SourceContext
import org.apache.flink.streaming.api.scala.StreamExecutionEnvironment
import org.apache.flink.streaming.api.scala._

object WriteIntoKafkaUseKafkaSink {
  def main(args: Array[String]): Unit = {
    System.out.println("use command as: ")
    System.out.println("./bin/flink run --class class com.huawei.bigdata.flink.examples.WriteIntoKafkaUseKafkaSink" +
      " /opt/test.jar --topic topic-test -bootstrap.servers xxx.xxx.xxx.xxx:21005")
    System.out.println(
      "./bin/flink run --class class com.huawei.bigdata.flink.examples.WriteIntoKafkaUseKafkaSink /opt/test.jar --topic"
        + " topic-test -bootstrap.servers xxx.xxx.xxx.xxx:21007 --security.protocol SASL_PLAINTEXT"
        + " --sasl.kerberos.service.name kafka --kerberos.domain.name hadoop.系统域名");
    System.out.println("******************************************************************************************")
    System.out.println("<topic> is the kafka topic name")
    System.out.println("<bootstrap.servers> is the ip:port list of brokers")
    System.out.println("******************************************************************************************")
    val env = StreamExecutionEnvironment.getExecutionEnvironment
    env.setParallelism(1)
    val paraTool = ParameterTool.fromArgs(args)
    val messageStream: DataStream[String] = env.addSource(new SimpleStringGenerator2)
    val build = KafkaSink.builder[String]
      .setBootstrapServers(paraTool.get("bootstrap.servers"))
      .setRecordSerializer(KafkaRecordSerializationSchema.builder[String]
        .setValueSerializationSchema(new SimpleStringSchema())
        .setTopic(paraTool.get("topic"))
        .build)
      .setKafkaProducerConfig(paraTool.getProperties)
      .build
    messageStream.sinkTo(build)
    env.execute
  }
}

class SimpleStringGenerator2 extends SourceFunction[String] {
  var running = true
  var i = 0

  override def run(ctx: SourceContext[String]) {
    while (running) {
      ctx.collect("element-" + i)
      i += 1
      Thread.sleep(1000)
    }
  }

  override def cancel() {
    running = false
  }
}

File: src/flink-examples/flink-examples-security/FlinkKafkaJavaExample/src/main/java/com/huawei/bigdata/flink/examples/WriteIntoKafkaUseKafkaSink.java
/*
 * Copyright (c) Huawei Technologies Co., Ltd. 2012-2020. All rights reserved.
 */

package com.huawei.bigdata.flink.examples;

import org.apache.flink.api.common.serialization.SimpleStringSchema;
import org.apache.flink.api.java.utils.ParameterTool;
import org.apache.flink.connector.kafka.sink.KafkaRecordSerializationSchema;
import org.apache.flink.connector.kafka.sink.KafkaSink;
import org.apache.flink.streaming.api.datastream.DataStream;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
import org.apache.flink.streaming.api.functions.source.SourceFunction;

/**
 * @since 8.0.2
 */
public class WriteIntoKafkaUseKafkaSink {
    public static void main(String[] args) throws Exception {
        System.out.println("use command as: ");
        System.out.println(
                "./bin/flink run --class com.huawei.bigdata.flink.examples.WriteIntoKafka"
                        + " /opt/test.jar --topic topic-test -bootstrap.servers xxx.xxx.xxx.xxx:21005");
        System.out.println(
                "./bin/flink run --class com.huawei.bigdata.flink.examples.WriteIntoKafka /opt/test.jar --topic"
                        + " topic-test -bootstrap.servers xxx.xxx.xxx.xxx:21007 --security.protocol SASL_PLAINTEXT"
                        + " --sasl.kerberos.service.name kafka --kerberos.domain.name hadoop.系统域名");
        System.out.println(
                "******************************************************************************************");
        System.out.println("<topic> is the kafka topic name");
        System.out.println("<bootstrap.servers> is the ip:port list of brokers");
        System.out.println(
                "******************************************************************************************");
        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
        env.setParallelism(1);
        ParameterTool paraTool = ParameterTool.fromArgs(args);
        DataStream<String> messageStream = env.addSource(new WriteIntoKafka.SimpleStringGenerator());
        KafkaSink<String> build = KafkaSink.<String>builder()
                .setBootstrapServers(paraTool.get("bootstrap.servers"))
                .setRecordSerializer(
                        KafkaRecordSerializationSchema.builder()
                                .setValueSerializationSchema(new SimpleStringSchema())
                                .setTopic(paraTool.get("topic"))
                                .build())
                .setKafkaProducerConfig(paraTool.getProperties())
                .build();
        messageStream.sinkTo(build);
        env.execute();
    }

    /**
     * @since 8.0.2
     */
    public static class SimpleStringGenerator implements SourceFunction<String> {
        private static final long serialVersionUID = 2174904787118597072L;
        boolean running = true;
        long i = 0;

        @Override
        public void run(SourceContext<String> ctx) throws Exception {
            while (running) {
                ctx.collect("element-" + (i++));
                Thread.sleep(1000);
            }
        }

        @Override
        public void cancel() {
            running = false;
        }
    }
}
